setup.cfg

```diff
@@ -35,12 +35,20 @@ setup_requires =
     pytest-runner
 install_requires =
     makefun>=1.5.0
+    # we're using typing_extensions even for new versions of python, so we don't pollute our code
+    # with try/except every time we use new feature.
+    # typing_extensions will use version from typing module if it detects new enough python.
+    typing-extensions>=4.2;python_version>='3.6'
```

Comment:
I prefer having no additional dependency if the python version is recent enough. So I would rather suggest to bound this with an upper python version, and to keep the try/except in the code below.